keras_core/ops/function.py

```diff
@@ -46,10 +47,20 @@ class Function(Operation):
     def __init__(self, inputs, outputs, name=None):
         super().__init__(name=name)
 
+        if backend() == "tensorflow":
```

Comment:
Can we do this in `keras_core/backend/tensorflow/layer.py::TFLayer.__init__` instead?

Reply:
I don't think so, because we can't turn off tracking overall for functional models. We only want to turn it off for the input/output structs here to avoid the tf bug. Essentially we need a way to run a hook inside `Function.__init__`.

The nice way to avoid all this would be to use `Tracker` for our tf tracking. This could save a lot of bugs I think, but is more of an effort. https://github.com/keras-team/keras-core/issues/934#issuecomment-1728464445

Reply:
Maybe we could just not depend on `AutoTrackable` (just `Trackable`) and then manually track everything according to our own `Tracker`.

Reply:
Yeah, I think this would take some work, but really improve the overall state if we did.